donkeycar/parts/datastore_v2.py

```diff
@@ -0,0 +1,367 @@
+import atexit
+import json
+import mmap
+import os
+import time
+from collections import namedtuple
+from pathlib import Path
+
+
+class Seekable(object):
+    '''
+    A seekable file reader, writer which deals with newline delimited records. \n
+    This reader maintains an index of line lengths, so seeking a line is a O(1) operation.
+    '''
+
+    def __init__(self, file, method='a+', line_lengths=list()):
+        self.method = method
+        self.line_lengths = list()
+        self.file = open(file, self.method)
+        if len(line_lengths) <= 0:
+            self._read_contents()
+        else:
+            self.line_lengths.extend(line_lengths)
+
+    def _read_contents(self):
+        self.line_lengths.clear()
+        self.file.seek(0)
+        contents = self.file.readline()
+        while len(contents) > 0:
```

Comment:
Looks like this treats first empty line as EOF, is that correct?  I think that is fine for our purposes, but maybe we add that in the class' comment.